controllers/reservations.js

```diff
@@ -65,6 +65,14 @@ exports.addReservation = async (req, res, next) => {
     req.body.user = req.user.id;
     req.body.shop = req.params.shopId;
 
+    const isScheduleAvailable = await checkScheduleAvailability(req);
+    if (isScheduleAvailable.statusCode != 200) {
+      return res.status(isScheduleAvailable.statusCode).json({
+        success: false,
+        message: isScheduleAvailable.message,
+      });
+    }
```

Comment:
should handle status code with other way?

Reply:
The `return res.status` can only be used in the controller function. I tried using it in `checkScheduleAvailability()` function and got an error about cannot set headers. IDK how to handle the case so I choose this one as sometimes I use it in my code workspace.